ledger-core/virtual/integration/deduplication/constructor_test.go

```diff
@@ -139,3 +141,206 @@ func TestConstructor_SamePulse(t *testing.T) {
+		},
+		{
+			Name:       "status is empty and pending is known",
+			TestRailID: "C5012",
+			VState: payload.VStateReport{
+				Status:              payload.Empty,
+				OrderedPendingCount: 1,
+				ProvidedContent:     &payload.VStateReport_ProvidedContentBody{},
+			},
+			VStateSendBefore: true,
+			VDelegatedCall: &payload.VDelegatedCallRequest{
+				Callee:       payload.Reference{},
+				CallOutgoing: payload.Reference{},
+				CallFlags:    payload.BuildCallFlags(isolation.Interference, isolation.State),
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			t.Log(test.TestRailID)
+			if test.Skipped != "" {
+				t.Skip(test.Skipped)
+			}
+
+			mc := minimock.NewController(t)
+
+			server, ctx := utils.NewUninitializedServer(nil, t)
+			defer server.Stop()
+
+			executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
+
+			runnerMock := logicless.NewServiceMock(ctx, mc, nil)
+			server.ReplaceRunner(runnerMock)
+			server.Init(ctx)
+
+			var (
+				class         = gen.UniqueGlobalRef()
+				previousPulse = server.GetPulse().PulseNumber
+				outgoingLocal = gen.UniqueLocalRefWithPulse(previousPulse)
+				outgoing      = reference.NewRecordOf(server.GlobalCaller(), outgoingLocal)
+				object        = outgoing
+			)
+
+			test.VState.Object = object
+			test.VState.OrderedPendingEarliestPulse = previousPulse
+			if test.VState.ProvidedContent != nil {
+				test.VState.ProvidedContent.OrderedQueue = []reference.Global{outgoing}
+			}
+
+			if test.VFindCall != nil {
+				test.VFindCall.Callee = outgoing
+				test.VFindCall.Outgoing = outgoing
+			}
+
+			if test.VDelegatedCall != nil {
+				test.VDelegatedCall.Callee = outgoing
+				test.VDelegatedCall.CallOutgoing = outgoing
+			}
+
+			{
+				requestResult := requestresult.New([]byte("123"), gen.UniqueGlobalRef())
+				requestResult.SetActivate(gen.UniqueGlobalRef(), class, []byte("234"))
+
+				executionMock := runnerMock.AddExecutionMock(outgoing.String())
+				executionMock.AddStart(nil, &execution.Update{
+					Type:   execution.Done,
+					Result: requestResult,
+				})
+			}
+
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+			typedChecker.VCallResult.Set(func(result *payload.VCallResult) bool {
+				assert.Equal(t, []byte("123"), result.ReturnArguments)
+				return false
+			})
+			typedChecker.VStateRequest.Set(func(request *payload.VStateRequest) bool {
+				VStateReportDone := server.Journal.WaitStopOf(&handlers.SMVStateReport{}, 1)
+				server.SendPayload(ctx, &test.VState)
+				testutils.WaitSignalsTimed(t, 10*time.Second, VStateReportDone)
+
+				assert.Equal(t, object, request.Object)
+
+				return false
+			})
+			typedChecker.VFindCallRequest.Set(func(request *payload.VFindCallRequest) bool {
+				if test.VFindCall == nil {
+					t.Fatal("unreachable")
+				}
+
+				assert.Equal(t, outgoing, request.Outgoing)
+				assert.Equal(t, object, request.Callee)
+
+				test.VFindCall.LookedAt = request.LookAt
```

Comment:
this field should be checked, not just copied, this whole test is around VFindCallRequest/VFindCallResponse, so all fields should be checked.

Reply:
fixed